### ISSUE TITLE
nom 4.0.0-alpha2: many0!(alt!()) on CompleteByteSlice returns unexpected errors

### ISSUE THREAD

**Issue by vthriller** [Geal/nom#667]
```
#[macro_use]
extern crate nom;
use nom::alpha;
use nom::types::CompleteByteSlice;

named!(foo <CompleteByteSlice, Vec<CompleteByteSlice>>, many0!(alt!(alpha | is_a!("_"))));

fn main() {
    println!("{:?}", foo(CompleteByteSlice(&b""[..])));
    println!("{:?}", foo(CompleteByteSlice(&b"loremipsum"[..])));
    println!("{:?}", foo(CompleteByteSlice(&b"lorem_ipsum"[..])));
    println!("{:?}", foo(CompleteByteSlice(&b"_lorem_ipsum"[..])));
    println!("{:?}", foo(CompleteByteSlice(&b"!@#$"[..])));
}
```

Actual output:

```
Err(Error(Code(CompleteByteSlice([]), Many0)))
Err(Error(Code(CompleteByteSlice([]), Many0)))
Err(Error(Code(CompleteByteSlice([]), Many0)))
Err(Error(Code(CompleteByteSlice([]), Many0)))
Ok((CompleteByteSlice([33, 64, 35, 36]), []))
```

Expected: every return value to be `Ok()` of some sort.

**Comment by Geal** [comment]
alright, I found the cause. Thank you for catching that, there are some cases I forgot here and there

**Comment by Geal** [comment]
this should be fixed by 968ae9f. I added a unit test with the example you provided in 62a0e75

**Comment by vthriller** [comment]
Indeed it works now in [my promql parser](https://github.com/vthriller/promql). Thanks!.
